fms/modules/feedforward.py

```diff
@@ -486,58 +505,98 @@ def reset_parameters(self):
                 std=0.02,
             )
 
+    def is_parameter_initialized(self, param: nn.Parameter):
+        return param is not None and isinstance(param, nn.Parameter)
+
     def to_tp(self, group: ProcessGroup) -> "TPConditionalFeedForward":
         return TPConditionalFeedForward.import_module(self, group)
 
-    def forward(self, x: torch.Tensor, expert_indices: torch.Tensor) -> torch.Tensor:
+    def swiglu(self, x, alpha: float = 1.702, limit: float = 7.0):
+        x_glu, x_linear = x[..., ::2], x[..., 1::2]
+        # Clamp the input values
+        x_glu = x_glu.clamp(min=None, max=limit)
+        x_linear = x_linear.clamp(min=-limit, max=limit)
+        out_glu = x_glu * torch.sigmoid(alpha * x_glu)
+        # Note we add an extra bias of 1 to the linear layer
+        return out_glu * (x_linear + 1)
+
+    def forward(
```

Comment:
this is not reusing existing code well:
1. the `use_bias` condition should only affect `+ mlp1_bias` and `+ mlp2_bias`, not the whole computation
2. the swiglu should use the default pytorch implementation unless you have the limit passed in from higher up in the stack
3. the weighted sum of experts happens in MOEFeedFoward already